app/workers/billing_worker.rb

```diff
@@ -25,35 +38,69 @@ def on_complete(status, options)
     end
   end
 
+  SPARSING_RATE = (2.0/3).freeze # in seconds/transaction
+
+  def self.sparsing_rate
+    SPARSING_RATE
+  end
+
   # @param [Account] provider
   # @param [Time] billing_date
   # @param [ActiveRecord::Relation] buyers_scope
   def self.enqueue(provider, billing_date, buyers_scope = nil)
+    payment_gateway = PaymentGateway.find(provider.payment_gateway_type)
+    needs_lock = payment_gateway&.need_lock?
+    needs_sparsing = payment_gateway&.need_sparsing?
+
     with_billing_batch(provider.id, billing_date) do
+      count = 0
       provider.buyer_accounts.select(:id, :provider_account_id).merge(buyers_scope).find_in_batches do |group|
-        group.each { |buyer| enqueue_for_buyer(buyer, billing_date) }
+        group.each do |buyer|
+          options = { needs_lock: needs_lock }
+
+          if needs_sparsing
+            options[:delay] = (sparsing_rate * count).seconds
+            count += 1
+          end
+
+          enqueue_for_buyer(buyer, billing_date, **options)
+        end
       end
     end
   end
 
-  def self.enqueue_for_buyer(buyer, billing_date)
-    time = billing_date.to_s(:iso8601)
-    perform_async(buyer.id, buyer.provider_account_id, time)
-  end
-
   def self.with_billing_batch(provider_id, billing_date, &block)
     batch = Sidekiq::Batch.new
     batch.description = "Billing (id: #{provider_id})"
     batch.on(:complete, BillingWorker::Callback, batch_id: batch.bid, account_id: provider_id, billing_date: billing_date)
     batch.jobs(&block)
   end
 
+  def self.enqueue_for_buyer(buyer, billing_date, needs_lock: false, delay: 0.second)
```

Comment:
Rubocop and I agree on this 😄 It sounds better to me 😄 
![image](https://user-images.githubusercontent.com/11318903/51188236-d6143b80-18dd-11e9-9d57-dd78b22cf425.png)